api/backend.proto

```diff
@@ -54,28 +54,14 @@ option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
   // https://github.com/grpc-ecosystem/grpc-gateway/blob/master/examples/proto/examplepb/a_bit_of_everything.proto
 };
 
-// Configuration for a GRPC Match Function
-message GrpcFunctionConfig {
-  string host = 1;
-  int32 port = 2;
-}
-
-// Configuration for a REST Match Function.
-message RestFunctionConfig {
-  string host = 1;
-  int32 port = 2;
-}
```

Comment:
@jeremyje - Is there a recommendation - or a canonical way to accept a GRPC endpoint address? Between passing in entire address v/s accepting host, port and creating the address when generating a client, which one is preferred?

Reply:
Splitting out host and port seems like the most generic way. I just looked at the java example.

```java
public RouteGuideClient(String host, int port) {
  this(ManagedChannelBuilder.forAddress(host, port).usePlaintext());
}
```

Reply:
I'll update this commit to split out the host and port, yet I think the change of using enum type instead of `oneof` is still valid. This could keep us from casting things around.